visualization/include/visualization/element.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <vector>
+#include <utility>
+#include <string>
+
+namespace visualization::builder {
+
+    class Element {
+    public:
+        friend class ElementBuilder;
```

Comment:
You could put a child class here. So you have Element::Builder` instead of ElementBuilder. Friend classes should be avoided as much as possible